CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.12)
+cmake_minimum_required(VERSION 3.20)
```

Comment:
Is there a CMake feature that requires the version bump?

Reply:
Yes, it is to be coherent with recent HDF5 and also from this cmake version the linking process to HDF5 (finding dependencies, packaging to transfer to consumer code the dependency) has been modified to be more reliable (feature https://cmake.org/cmake/help/v3.20/module/FindHDF5.html).